app-center-services/src/main/java/org/exoplatform/appcenter/search/ApplicationSearchConnector.java

```diff
@@ -0,0 +1,195 @@
+        }
+
+        long userIdentityId = getCurrentUserIdentityId(identityManager, currentUser);
+        if (userIdentityId < 0) {
+            throw new IllegalArgumentException("User identity id must be positive");
+        }
+        String esQuery = "";
+        if (StringUtils.isBlank(term)) {
+            esQuery = buildQueryStatement(userIdentityId, term, offset, limit,searchAllQueryFilePath );
+        }
+        else {
+            esQuery = buildQueryStatement(userIdentityId, term, offset, limit,searchQueryFilePath );
+
+        }
+        String jsonResponse = this.client.sendRequest(esQuery, this.index, this.searchType);
+        return buildResult(jsonResponse);
+    }
+
+
+    @SuppressWarnings("rawtypes")
+    public List<ApplicationSearchResult> buildResult(String jsonResponse) {
+        LOG.debug("Search Query response from ES : {} ", jsonResponse);
+        List<ApplicationSearchResult> results = new ArrayList<>();
+        JSONParser parser = new JSONParser();
+
+        Map json = null;
+        try {
+            json = (Map) parser.parse(jsonResponse);
+        } catch (ParseException e) {
+            throw new ElasticSearchException("Unable to parse JSON response", e);
+        }
+
+        JSONObject jsonResult = (JSONObject) json.get("hits");
+        if (jsonResult == null) {
+            return results;
+        }
+
+        JSONArray jsonHits = (JSONArray) jsonResult.get("hits");
+        for (Object jsonHit : jsonHits) {
+            try {
+                ApplicationSearchResult applicationSearchResult = new ApplicationSearchResult();
+                JSONObject jsonHitObject = (JSONObject) jsonHit;
+                JSONObject hitSource = (JSONObject) jsonHitObject.get("_source");
+                long id = parseLong(hitSource, "id");
+                String title = (String) hitSource.get("title");
+                String description = (String) hitSource.get("description");
+                JSONObject highlightSource = (JSONObject) jsonHitObject.get("highlight");
+                List<String> excerpts = new ArrayList<>();
+                if (highlightSource != null) {
+                    if (highlightSource.get("description") != null) {
+                        description = ((JSONArray) highlightSource.get("description")).get(0).toString();
+                        highlightSource.remove("description");
+                    }
+                    for (Object value : highlightSource.values()) {
+                        JSONArray excerptValue = (JSONArray) value;
+                        if (excerptValue != null && !excerptValue.isEmpty()) {
+                            excerpts.add(excerptValue.get(0).toString());
+                        }
+                    }
+                }
+                // Application
+                applicationSearchResult.setId(id);
+                applicationSearchResult.setDescription(description);
+                applicationSearchResult.setTitle(title);
```

Comment:
Don't we need the application image too ? I think it is mandatory to have it when searching on the Unified Search